core/templates/dev/head/components/CreateExplorationButtonDirective.js

```diff
@@ -44,6 +49,43 @@ oppia.directive('createExplorationButton', [function() {
           }, 150);
           return false;
         };
+
+        $scope.showCreationChoiceModal = function() {
+          var modalInstance = $modal.open({
+            templateUrl: 'modals/createExplorationOrCollection',
+            backdrop: true,
+            controller: [
+              '$scope', '$modalInstance', function($scope, $modalInstance) {
+                $scope.choseExploration = false;
```

Comment:
I don't think you need these variables -- it's a bit complicated. Why not directly call $scope.createNewExploration() etc. from within the modal? Would that work?